File: atcoder/abc172/d.py
from collections import defaultdict
from functools import lru_cache
N = int(input())
spf = [_ for _ in range(N + 1)]
i = 2
while i * i <= N:
    if spf[i] == i:
        for j in range(i * i, N + 1, i):
            if spf[j] == j:
                spf[j] = i
    i += 1


@lru_cache(maxsize=None)
def factorize(n):
    m = defaultdict(int)
    while n != 1:
        m[spf[n]] += 1
        n = n // spf[n]
    res = 1
    for i in m:
        res = res * (m[i] + 1)
    return res


res = 0
for i in range(1, N + 1):
    res += i * factorize(i)
print(res)

File: atcoder/abc172/d.cpp
#include <bits/stdc++.h>
using namespace std;
#define int long long

int factorize(vector<int> &spf, int n)
{
    map<int, int> mp;
    while (n != 1)
    {
        mp[spf[n]]++;
        n = n / spf[n];
    }
    int res = 1;
    for (auto it = mp.begin(); it != mp.end(); it++)
    {
        res = res * (it->second + 1);
    }
    return res;
}

signed main()
{
    int N;
    cin >> N;
    vector<int> spf(N + 1, 0);
    for (int i = 0; i < N + 1; i++)
    {
        spf[i] = i;
    }
    int i = 2;
    while (i * i <= N)
    {
        if (spf[i] == i)
        {
            for (int j = i * i; j < N + 1; j += i)
            {
                if (spf[j] == j)
                {
                    spf[j] = i;
                }
            }
        }
        i++;
    }
    int ans = 0;
    for (int i = 1; i < N + 1; i++)
    {
        ans = ans + i * factorize(spf, i);
    }
    cout << ans << endl;
}